Alderis/UIColor+HBAdditions.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  UIColor+HBAdditions.swift
+//  Alderis
+//
+//  Created by Ryan Nair on 10/5/20.
+//  Copyright © 2020 HASHBANG Productions. All rights reserved.
+//
+
+import UIKit
+
+extension UIColor {
+	
+	/// Initializes and returns a color object using data from the specified object.
+	///
+	/// The value is expected to be one of:
+	///
+	/// * An array of 3 or 4 integer RGB or RGBA color components, with values between 0 and 255 (e.g.
+	///   `@[ 218, 192, 222 ]`)
+	/// * A CSS-style hex string, with an optional alpha component (e.g. `#DAC0DE` or `#DACODE55`)
+	/// * A short CSS-style hex string, with an optional alpha component (e.g. `#DC0` or `#DC05`)
+	///
+	/// @param value The object to retrieve data from. See the discussion for the supported object
+	/// types.
+	/// @returns An initialized color object. The color information represented by this object is in the
+	/// device RGB colorspace.
+	
+	@objc convenience init?(hbcp_propertyListValue value: Any?) {
+		
+			if value == nil {
```

Comment:
Since we need to use the typecast value immediately after we check for it, this block could be restructured with `if let`. For instance:

```swift
if let array = value as? [FixedWidthInteger], array.count == 3 || array.count == 4 {
	// …
} else if let string = value as? String {
	// …
}
return nil
```

Using `[FixedWidthInteger]` also makes sure we won’t crash if the type *is* an array, but *not* of a type that can be cast to `CGFloat`. In that case it’ll fall through to the final `return nil`.

The `,` in the first if-let allows adding extra statements to the check (Swift needs a comma instead of `&&` there to logically separate it from the following statement).